### ISSUE TITLE
Spirit Consumption

### ISSUE THREAD

**Issue by joesnellpdx** [JDRF/spirit#178]
@patrickmckowen 
@kevinmpowell 
@jamesmelzer 

James, Kevin,

I wanted to get out in front of the how we plan on consuming Spirit. Patrick mentioned bringing this up in a planning session - but I decided to get the questions and thoughts into a project issue so we have a place to discuss. 

I apologize if any of my assumptions are off -- one more reason for this dialogue.

How does Eightshapes recommend consuming the design system? 

**The system:**
1. Are we intending to use npm package management ( for versions )? If not, how do you recommend sites / apps consume the system?
2. If we use NPM (or depending on how we are consuming the system), is the assumption that the files will be compiled per site / app after pulled in via npm?
3. What is the timeline for creating an endpoint for consumption? Working on proofs of concept, ideally we could begin on this sooner than later.

**The components:**
1. Do you have examples of how a site / app would use njk macro to output markup or is it expected for users to manually recreate markup? 
2. Are macros and macro usage part of the doc site?
3. Any general tips here?

**Comment by kevinmpowell** [comment]
> Are we intending to use npm package management ( for versions )? If not, how do you recommend sites / apps consume the system?
Yes, we're using npm. I believe @patrickmckowen said we'd be publishing to (npmjs.com)[npmjs.com]. 

> If we use NPM (or depending on how we are consuming the system), is the assumption that the files will be compiled per site / app after pulled in via npm?
If the consuming product is able to use `scss` then yes, they'd `@import` what they want from us and compile their own stylesheet. The node package we publish will contain a compiled `spirit.css`, `spirit.svg`, and possibly `spirit.js` file that can be used directly.

> What is the timeline for creating an endpoint for consumption? Working on proofs of concept, ideally we could begin on this sooner than later.
The `doc-site` is already a consumer of the `library`. Though it's locally `npm link`'d during development, it will build based on the published npmjs.com version of the library.


> Do you have examples of how a site / app would use njk macro to output markup or is it expected for users to manually recreate markup?
It is expected that users will manually recreate markup. Nunjucks is wholly a "design system core team" concern. Others need not know what we're using "under the covers."

> Are macros and macro usage part of the doc site?
Not typically.

> Any general tips here?
If you're asking about my PR comments regarding macro APIs, its more about core team maintenance and consistency, but not as high a priority as having well structured markup and styles as the output. We need not get hung up on creating the "perfect nunjucks macros" as long as we're adhering to the same coding style and understand the maintenance costs of what we're constructing with nunjucks.

**Comment by patrickmckowen** [comment]
@joesnellpdx @kevinmpowell what are your npmjs usernames so I can add you to https://www.npmjs.com/org/jdrfhq

**Comment by kevinmpowell** [comment]
`kevinmpowell`

**Comment by joesnellpdx** [comment]
@patrickmckowen 

joesnellpdx

@kevinmpowell Why does this DS include macros? Just for creating the doc-site / libary?
https://comet.discoveryeducation.com/components/callouts.html

**Comment by kevinmpowell** [comment]
@joesnellpdx Discovery was a rare exception in that they tooled up their product suite to actually use Nunjucks. It's certainly an option, but fairly rare. In my experience most product teams are already using Framework X or CMS Solution Y and telling them "use this new design system templating language in your product" is often a good way to get pushback and a lack of adoption by product teams.

**Comment by joesnellpdx** [comment]
@kevinmpowell Thanks! Your answers are super helpful!

@jamesmelzer I'm keeping this task open for reference. If that is an issue, feel free to close.
